airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationDefinitionsHandler.java

```diff
@@ -121,12 +133,21 @@ public DestinationDefinitionRead getDestinationDefinition(DestinationDefinitionI
         configRepository.getStandardDestinationDefinition(destinationDefinitionIdRequestBody.getDestinationDefinitionId()));
   }
 
-  public DestinationDefinitionRead createDestinationDefinition(DestinationDefinitionCreate destinationDefinitionCreate)
+  public DestinationDefinitionReadWithJobInfo createDestinationDefinition(DestinationDefinitionCreate destinationDefinitionCreate)
       throws JsonValidationException, IOException {
-    imageValidator.assertValidIntegrationImage(destinationDefinitionCreate.getDockerRepository(),
-        destinationDefinitionCreate.getDockerImageTag());
+    // Validates that the docker image exists and can generate a compatible spec by running a getSpec
+    // job on the provided image and checking that getOutput() worked.
+    SynchronousResponse<ConnectorSpecification> response = null;
+    try {
+      response = specFetcher.executeWithResponse(DockerUtils.getTaggedImageName(destinationDefinitionCreate.getDockerRepository(), destinationDefinitionCreate.getDockerImageTag()));
+      Preconditions.checkNotNull(response, "Get Spec job returned null response");
+      Preconditions.checkState(response.isSuccess(), "Get Spec job failed.");
+      Preconditions.checkNotNull(response.getOutput(), "Get Spec job return null spec");
+    } catch (NullPointerException e) {
```

Comment:
imo i don't think we should batching the runtime exceptions here (or in the 3 other cases in this PR). if we throw an exception in this clause it means that something has gone sideways with our job infrastructure. passing this information externally in the API doesn't add much. it's not going to be fixable by the user. I think this is the case where we let the runtime exceptions bubble up just end up returning a 500.

Reply:
There are 4 endpoints affected by these error-handling changes.  We're discussing changing the contract to return 200 OK but with error details in the json response, instead of throwing a 422 with an exception.  That results in changes to the exception handling / checking structure in this area, as well.  NPE and IllegalStateException would still need to be caught, to be converted to normal program flow instead of using asserts.  The JobInfo and error details / exception details should remain appropriately carried through to the new error handling strategy as well, to maintain visibility into exactly which failure occurred.  I will modify the code a bit more to accomplish this.

Reply:
Made the change for sending ok with failure details.